### ISSUE TITLE
prefer_single_quotes should lint without looking into string interpolation

### ISSUE THREAD

**Issue by a14n** [dart-lang/linter#812]
`"hello ${'world'}"` should be linted.

**Comment by aemino** [comment]
At the moment, the `prefer_single_quotes` rule seems to intentionally allow that syntax. The reasoning is laid out [here](https://github.com/dart-lang/linter/blob/3bfa6428b26c76cfd6d4da4cdeb2dd7581be0081/lib/src/rules/prefer_single_quotes.dart#L19-L22).

**Comment by a14n** [comment]
Thanks, it can make sense for some code. And I guess that's why expressions in string interpolation are not linted:

```dart
    "${""}"; // OK
    '${''}'; // OK
    '${'${'${'${''}'}'}'}'; // OK
    '${"${'${"${''}"}'}"}'; // OK
```
